lib/typed_ecto_schema/ecto_type_mapper.ex

```diff
@@ -109,6 +114,23 @@ defmodule TypedEctoSchema.EctoTypeMapper do
   ## Type Transformation Helpers
   ##
 
+  @spec disjunction_typespec(list(atom())) :: Macro.t()
+  defp disjunction_typespec([sole_item]) when is_atom(sole_item) do
+    sole_item
+  end
+
+  defp disjunction_typespec([first, last]) when is_atom(first) and is_atom(last) do
+    quote do
+      unquote(first) | unquote(last)
+    end
+  end
+
+  defp disjunction_typespec([head | tail]) when is_atom(head) do
+    quote do
+      unquote(head) | unquote(disjunction_typespec(tail))
+    end
+  end
+
```

Comment:
```suggestion

  defp disjunction_typespec(_) do
    quote do
      any()
    end
  end
```

I tested this and the only problem I found was that if the user made a mistake, such as not declaring the values or something around these lines, the code was throwing a cryptic error such as "no pattern match" and was omitting the nicely printed error Ecto gives us:

```
== Compilation error in file lib/test.ex ==
** (ArgumentError) Ecto.Enum types must have a values option specified as a list of atoms. For example:

    field :my_field, Ecto.Enum, values: [:foo, :bar]
```

So I just added this fallback that just defaults to `any()`. It is not great, but makes this lib less intrusive.

But apart from this detail, great job!